internal/db/repos/instance.go

```diff
@@ -69,13 +69,16 @@ func (r *InstanceRepository) UpdateByID(ctx context.Context, id uint, instance *
 }
 
 // UpdateByName updates an instance by its name. Only non-zero fields in the instance parameter will be updated.
-// See UpdateByID method for details on how GORM handles field updates.
 func (r *InstanceRepository) UpdateByName(ctx context.Context, name string, instance *models.Instance) error {
 	return r.db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
-		if err := tx.Model(&models.Instance{}).
-			Where(&models.Instance{Name: name}).
-			Updates(instance).Error; err != nil {
-			return fmt.Errorf("failed to update instance by name: %w", err)
+		result := tx.Model(&models.Instance{}).
+			Where("name = ? AND deleted_at IS NULL", name).
```

Comment:
what value this will bring?
the previous code `Where(&models.Instance{Name: name}).` seems much cleaner and not used hard coded field names